tests/e2e/test_parameterized_queries.py

```diff
@@ -0,0 +1,382 @@
+            date_col DATE,
+            timestamp_col TIMESTAMP
+            ) USING DELTA
+        """
+
+        with self.connection() as conn:
+            with conn.cursor() as cursor:
+                cursor.execute(query)
+
+    def compute_says_it_doesnt_support_native_params(self):
+        """we test against dogfood. dogfood exposes a protocol version that suggests it can't handle
+        native queries, even though it does. for test coverage, we can mock the local protocol check.
+        but we want to remove this mock once dogfood properly advertises its version.
+        """
+
+        with self.connection() as conn:
+            return (
+                conn.protocol_version
+                < ttypes.TProtocolVersion.SPARK_CLI_SERVICE_PROTOCOL_V8
+            )
+
+    @contextmanager
+    def conditional_protocol_patch(self, bypass=False):
+        """This fixture will be removed once dogfood advertises its protocol version correctly.
+        
+        Note that there is an equivalent patch in sqlalchemy/test/test_suite.py which should be 
+        removed at the same time as this one. That one is encapsulated in a function called
+        start_protocol_patch()"""
+
+        if bypass:
+            yield None
+
+        if not self.compute_says_it_doesnt_support_native_params():
+            yield None
+
+        with patch(
+            "databricks.sql.client.Connection.server_parameterized_queries_enabled",
+            return_value=True,
+        ) as mock_parameterized_queries_enabled:
+            try:
+                yield mock_parameterized_queries_enabled
+            finally:
+                pass
+
+    def _inline_roundtrip(self, params: dict):
+        target_column = self.inline_type_map[type(params.get("p"))]
+        INSERT_QUERY = f"INSERT INTO pysql_e2e_inline_param_test_table (`{target_column}`) VALUES (%(p)s)"
+        SELECT_QUERY = f"SELECT {target_column} `col` FROM pysql_e2e_inline_param_test_table LIMIT 1"
+        DELETE_QUERY = "DELETE FROM pysql_e2e_inline_param_test_table"
+
+        with self.connection(extra_params={"use_inline_params": True}) as conn:
+            with conn.cursor() as cursor:
+                cursor.execute(INSERT_QUERY, parameters=params)
+            with conn.cursor() as cursor:
+                to_return = cursor.execute(SELECT_QUERY).fetchone()
+            with conn.cursor() as cursor:
+                cursor.execute(DELETE_QUERY)
+
+        return to_return
+
+    def _native_roundtrip(
+        self, parameters: Union[Dict, List[Dict]], bypass_patch=False
+    ):
+        with self.connection(extra_params={"use_inline_params": False}) as conn:
+            with conn.cursor() as cursor:
+                with self.conditional_protocol_patch(bypass_patch):
+                    cursor.execute(self.NATIVE_QUERY, parameters=parameters)
+                    return cursor.fetchone()
+
+    def _get_one_result(self, approach: ParameterApproach, params, bypass_patch=False):
```

Comment:
`bypass_patch` is only needed when another test is already patching the protocol version.